src/test/java/com/fasterxml/jackson/databind/deser/jdk/BigDecimalForFloatDisabled3133Test.java

```diff
@@ -0,0 +1,63 @@
+package com.fasterxml.jackson.databind.deser.jdk;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import static com.fasterxml.jackson.databind.BaseMapTest.jsonMapperBuilder;
+import static com.fasterxml.jackson.databind.BaseTest.a2q;
+
+/**
+ * Unit test proving that below issue is fixed.
+ * <p>
+ * [databind#3133] Map deserialization results in different numeric classes based on json
+ * ordering (BigDecimal / Double) when used in combination with @JsonSubTypes
+ */
+public class BigDecimalForFloatDisabled3133Test
+{
+    @JsonTypeInfo(
+            use = JsonTypeInfo.Id.NAME,
+            include = JsonTypeInfo.As.PROPERTY,
+            property = "type")
+
+    @JsonSubTypes({
+            @JsonSubTypes.Type(value = TestMapContainer.class, name = "MAP"),
+    })
+    interface TestJsonTypeInfoInterface { }
+
+    static class TestMapContainer implements TestJsonTypeInfoInterface {
+
+        private Map<String, ? extends Object> map = new HashMap<>();
+
+        public Map<String, ? extends Object> getMap() {
+            return map;
+        }
+
+        public void setMap(Map<String, ? extends Object> map) {
+            this.map = map;
+        }
+    }
+
+    private final ObjectMapper mapper = jsonMapperBuilder()
+            .disable(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS)
+            .build();
+
+    @Test
+    public void testDeserializeWithDifferentOrdering() throws Exception {
+        // case 1 : type first
+        String ordering1 = a2q("{'type': 'MAP','map': { 'doubleValue': 0.1 }}");
+        TestMapContainer model1 = mapper.readValue(ordering1, TestMapContainer.class);
+        Assertions.assertTrue(model1.getMap().get("doubleValue") instanceof Double);
```

Comment:
I think this would be better as `assertEquals(0.1d, model1.getMap().get("doubleValue"))`

Reply:
At that time, I was unsure whether to closely align with the test case from #3133.
In hindsight, I should've gone with your suggestion. Thankss 🙏🏼